publication-commons/src/test/java/cucumber/permissions/file/FileScenarioContext.java

```diff
@@ -41,9 +44,15 @@ public FileEntry getFileEntry() {
     }
 
     public UserInstance getCurrentUserInstance() {
-        return UserInstance.create(userContext.userIdentifier, userContext.customerId, userContext.personCristinId,
-                                   userContext.accessRights.stream().toList(),
-                                   userContext.topLevelOrgCristinId);
+        if (userContext.userClientType == UserClientType.INTERNAL) {
+            return UserInstance.create(userContext.userIdentifier, userContext.customerId, userContext.personCristinId,
+                                       userContext.accessRights.stream().toList(),
+                                       userContext.topLevelOrgCristinId);
+        } else {
+            return UserInstance.createExternalUser(
+                new ResourceOwner(new Username(userContext.userIdentifier), userContext.topLevelOrgCristinId),
+                userContext.customerId);
+        }
```

Comment:
Tempted by a ternary that has methods `return isInternalUser(userContext) ? createInternalUser(userContext) : createExternalUser(userContext)`